src/analytics.js

```diff
@@ -95,9 +102,13 @@ function getEventProperties(eventName, target) {
 }
 
 
-// Assign a click event for any element with data-event-name set.
+// Assign a click event for any element with `data-event-name` set.
+// The element should also have the `data-event-label` and `data-event-block` custom attributes set.
 $('[data-event-name]').click(function() {
     const target = $(this).closest('[data-event-name]')[0]
     const eventName = target.getAttribute('data-event-name')
-    sendEvent(eventName, getEventProperties(eventName, target))
+    const eventProperties = getEventProperties(eventName, target)
+
+    // `getEventProperties` will return `null` if the event should not be sent.
+    if (eventProperties) sendEvent(eventName, eventProperties)
```

Comment:
Is there a case where we would still want to send an event even if there is no other properties attached to it?

Reply:
Unless the logic in `getEventProperties` explicitly returns `null`, it will return at a minimum an empty object, which is still considered truthy and will allow the event to be sent.